### DISCUSSION TITLE
Question: Pruning old webhook calls

### DISCUSSION THREAD

**Discussion by bomshteyn** [spatie/laravel-webhook-client#93] [Ideas]
Currently the webhook calls stay in the database even after being processed.

Maybe we can add a command similar to Pruning Job Batches: https://github.com/laravel/framework/blob/d786e741e15ad1afa723cef19f05dabc040be912/src/Illuminate/Queue/Console/PruneBatchesCommand.php

**Comment by Saifallak** [comment]
really need this feature,
but there is no way to know if the webhook call is processed or not on the database level.

**Comment by freekmurze** [comment]
I agree that we should support pruning out of the box.

I'd accept a PR that adds this. Make sure the amount of days before a web hook gets pruned is configurable.
Like @saifallak mentioned we'd also need a way to mark a webhook as processed.

**Comment by freekmurze** [comment]
I agree that we should support pruning out of the box. Like @saifallak mentioned we'd also need a way to mark a webhook as processed. 

I'd accept a PR that adds this. Make sure the amount of days before a processed web hook gets pruned is configurable.

**Comment by Saifallak** [comment]
I might delete webhook record in the end of job,..

as a work around for now,..

also will try to add PR for this, but not sure as my schedule is too busy At the moment

**Reply by ankurk91** [reply]
We can use Laravel's inbuilt model:prune command
https://laravel.com/docs/9.x/eloquent#pruning-models

**Reply by Saifallak** [reply]
> 

How?, you don't have any clues if it's processed or not.

**Reply by ankurk91** [reply]
I meant, this package can use the model prune feature, there is no need to intruduce a new artisan command
